### ISSUE TITLE
 assert (boxes1[:, 2:] >= boxes1[:, :2]).all() AssertionError

### ISSUE THREAD

**Issue by chenqq1** [ShoufaChen/DiffusionDet#60]
assert (boxes1[:, 2:] >= boxes1[:, :2]).all() AssertionError

**Comment by ShoufaChen** [comment]
Hi, I didn't meet this error in our experiments.

Did you use coco dataset or your custom data?

**Comment by chenqqqqqqq** [comment]
> Hi, I didn't meet this error in our experiments.
> 
> Did you use coco dataset or your custom data?

我用的是自己定义的数据集，这个报错在以下两种情况下都有发生：
1. 当我将AdamW优化器更换为一个最新的Lion优化器时这个错误会出现；
2. 我使用AdamW优化器，backbone选用resnet50时是可以对自己定义的数据集进行完整训练的。可是，当我将backbone换成resnet101时，迭代到第两万次时，这个错误又出现了！
不懂到底是为什么，难道是学习率每次都要做不同的调整吗？因为我的数据集成功跑通过其它很多模型，所以数据集是没有问题的。希望您可以给我解答一下。

**Comment by asferrer** [comment]
Hi! I'm having the same problem, did you managed to solve it?

**Comment by chenqqqqqqq** [comment]
Sorry, i didn't sovle it.
---- Replied Message ----
| From | Alejandro Sanchez ***@***.***> |
| Date | 07/02/2023 23:58 |
| To | ShoufaChen/DiffusionDet ***@***.***> |
| Cc | chenqqqqqqq ***@***.***>,
Comment ***@***.***> |
| Subject | Re: [ShoufaChen/DiffusionDet] assert (boxes1[:, 2:] >= boxes1[:, :2]).all() AssertionError (Issue #60) |

Hi! I'm having the same problem, did you managed to solve it?

**Comment by asferrer** [comment]
Hi! Thank you chenq, i realized that i could handle it by changing de optimizer to SGD instead of using the default AdamW. Hope it helps

**Comment by chenqqqqqqq** [comment]
yeah, I realize that this error is impacted by the optimizer. When i use AdamW, it runs normally, but when i change AdamW with other optimizer, such as the latest optimizer Lion, it goes wrong.


| |
陈萌
|
|
***@***.***
|
签名由网易邮箱大师定制
On 7/3/2023 17:18，Alejandro Sanchez ***@***.***> wrote：

Hi! Thank you chenq, i realized that i could handle it by changing de optimizer to SGD instead of using the default AdamW. Hope it helps
